src/ToricVarieties/cohomCalg/VanishingSets/constructors.jl

```diff
@@ -0,0 +1,32 @@
+#########################
+# 1: The Julia type for affine semigroups
+#########################
+
+@attributes mutable struct ToricVanishingSet
+    toric_variety::AbstractNormalToricVariety
+    scs::Vector{ShiftedCone}
+    i::fmpz
+    function ToricVanishingSet(toric_variety::AbstractNormalToricVariety, scs::Vector{ShiftedCone}, i::fmpz)
+        ambient_dimension = unique([ambient_dim(sc) for sc in scs])
+        if (length(ambient_dimension) > 1)
+            throw(ArgumentError("The ambient dimensions of the shifted cones must all agree."))
+        end
+        if (ambient_dimension[1] != rank(picard_group(toric_variety)))
+            throw(ArgumentError("The ambient dimensions of the shifted cones must match the rank as the picard group of the toric variety."))
+        end
+        if ((i < 0) || (i > dim(toric_variety)))
```

Comment:
A matter of taste, but note that Julia supports comparison chaining (not sure what the correct term for this is, i.e.: `a < b < c` is equivalent to `a < b && b < c`. So you could write:
```suggestion
        if !(0 <= i <= dim(toric_variety))
```
Of course you could also use a range:
```suggestion
        if !(i in 0:dim(toric_variety))
```

Reply:
Thank you for the pointer. Looks good!